src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -0,0 +1,265 @@
+            ITypeHintCommandFactory typeHintCommandFactory,
+            IConsoleUtilities consoleUtilities)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _orchestratorInteractiveService = orchestratorInteractiveService;
+            _cdkManager = cdkManager;
+            _systemCapabilityEvaluator = systemCapabilityEvaluator;
+            _cloudApplicationNameGenerator = cloudApplicationNameGenerator;
+            _awsUtilities = awsUtilities;
+            _awsClientFactory = awsClientFactory;
+            _awsResourceQueryer = awsResourceQueryer;
+            _projectParserUtility = projectParserUtility;
+            _commandLineWrapper = commandLineWrapper;
+            _cdkProjectHandler = cdkProjectHandler;
+            _deploymentBundleHandler = deploymentBundleHandler;
+            _templateMetadataReader = templateMetadataReader;
+            _deployedApplicationQueryer = deployedApplicationQueryer;
+            _typeHintCommandFactory = typeHintCommandFactory;
+            _consoleUtilities = consoleUtilities;
+        }
+
+        public Command BuildRootCommand()
+        {
+            // Name is important to set here to show correctly in the CLI usage help.
+            // Either dotnet-aws or dotnet aws works from the CLI. System.Commandline's help system does not like having a space with dotnet aws.
+            var rootCommand = new RootCommand {
+                Name = "dotnet-aws",
+                Description = "The AWS .NET deployment tool for deploying .NET applications on AWS."
+            };
+
+            rootCommand.Add(BuildDeployCommand());
+            rootCommand.Add(BuildListCommand());
+            rootCommand.Add(BuildDeleteCommand());
+
+            return rootCommand;
+        }
+
+        private Command BuildDeployCommand()
+        {
+             var deployCommand = new Command(
+                "deploy",
+                "Inspect, build, and deploy the .NET project to AWS using the recommended AWS service.")
+            {
+                _optionProfile,
+                _optionRegion,
+                _optionProjectPath,
+                _optionStackName,
+                _optionDiagnosticLogging,
+            };
+
+            deployCommand.Handler = CommandHandler.Create<string, string, string, string, bool, bool>(async (profile, region, projectPath, stackName, saveCdkProject, diagnostics) =>
+            {
+                try
+                {
+                    _toolInteractiveService.Diagnostics = diagnostics;
+
+                    var previousSettings = PreviousDeploymentSettings.ReadSettings(projectPath, null);
+
+                    var awsCredentials = await _awsUtilities.ResolveAWSCredentials(profile, previousSettings.Profile);
+                    var awsRegion = _awsUtilities.ResolveAWSRegion(region, previousSettings.Region);
+
+                    _commandLineWrapper.ConfigureProcess(async processStartInfo =>
```

Comment:
Maybe add a comment to explain what's going on here?

Actually, might be worth moving this to a dedicated external method to promote clarity and make it easier for dev's to grok.  Plus, it will allow this block of code to be reusable for Visual Studio implementation.

```
public static class CommandLineWrapperAwsCredsCotnext
{
   /// <summary>
   ///
   /// <para />
   /// TODO:  describe context, why with DI this strategy was used: ie plugins that add additional context
   /// to Command Execution, for example adding credentials.  Application has to inspect the environment to gather 
   /// credentials, or ask the user to provide them.  All of this must be done after DI container is built, so this strategy 
   ///  allows for late bound configuration.  
   /// </summary>
   public static void RegisterAwsCredsContext(
      this ICommandLineWrapper commandLineWrapper, 
      AWSCredentials creds,
      string region)
   {
      commandLineWrapper.ConfigureProcess(async processStartInfo => 
         // ...
   }
}
````

Reply:
yeah that's a good idea.